src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -350,51 +350,42 @@ private bool TryParseNamedEnum(
 
         private static ulong ConvertToUInt64(T value)
         {
-            switch (s_enumTypeCode)
-            {
-                case TypeCode.Int32 or TypeCode.UInt32: return Unsafe.As<T, uint>(ref value);
-                case TypeCode.Int64 or TypeCode.UInt64: return Unsafe.As<T, ulong>(ref value);
-                case TypeCode.Int16 or TypeCode.UInt16: return Unsafe.As<T, ushort>(ref value);
-                default:
-                    Debug.Assert(s_enumTypeCode is TypeCode.SByte or TypeCode.Byte);
-                    return Unsafe.As<T, byte>(ref value);
+            return s_enumTypeCode switch
+            {
+                TypeCode.Int32 => (ulong)(int)(object)value,
+                TypeCode.UInt32 => (uint)(object)value,
+                TypeCode.Int64 => (ulong)(long)(object)value,
+                TypeCode.UInt64 => (ulong)(object)value,
+                TypeCode.Int16 => (ulong)(short)(object)value,
+                TypeCode.UInt16 => (ushort)(object)value,
+                TypeCode.SByte => (ulong)(sbyte)(object)value,
+                _ => (byte)(object)value
```

Comment:
The previous code was asserting in the drain, and you're not.  If a new TypeCode were to be introduced, is the exception that comes out of this cast going to be useful enough?

Reply:
> The previous code was asserting in the drain, and you're not. If a new TypeCode were to be introduced, is the exception that comes out of this cast going to be useful enough?

if a new type code is added, the default will throw an InvalidCastException? we can only unbox byte out of a boxed byte.
PS: unfortunately, C# does not support statemens in pattern-match switch